test/integration/framework/cbor.go

```diff
@@ -0,0 +1,152 @@
+
+	newCBORSerializerInfo := func(creater runtime.ObjectCreater, typer runtime.ObjectTyper) runtime.SerializerInfo {
+		return runtime.SerializerInfo{
+			MediaType:        "application/cbor",
+			MediaTypeType:    "application",
+			MediaTypeSubType: "cbor",
+			Serializer:       cbor.NewSerializer(creater, typer),
+			StrictSerializer: cbor.NewSerializer(creater, typer, cbor.Strict(true)),
+			StreamSerializer: &runtime.StreamSerializerInfo{
+				Framer:     cbor.NewFramer(),
+				Serializer: cbor.NewSerializer(creater, typer, cbor.Transcode(false)),
+			},
+		}
+	}
+
+	dynamicAcceptContentTypes := dynamic.AcceptContentTypes
+	dynamicContentType := dynamic.ContentType
+	dynamicExtraSerializers := dynamic.ExtraSerializers
+	tb.Cleanup(func() {
+		dynamic.AcceptContentTypes = dynamicAcceptContentTypes
+		dynamic.ContentType = dynamicContentType
+		dynamic.ExtraSerializers = dynamicExtraSerializers
+	})
+	dynamic.AcceptContentTypes = "application/cbor;q=1,application/json;q=0.5"
+	dynamic.ContentType = "application/cbor"
+	dynamic.ExtraSerializers = append(dynamic.ExtraSerializers, newCBORSerializerInfo)
+
+	// Codecs for built-in types are constructed at package initialization time and read by
+	// value from REST storage providers.
+	codecs := map[*runtime.Scheme]*serializer.CodecFactory{
+		legacyscheme.Scheme:           &legacyscheme.Codecs,
+		metainternalscheme.Scheme:     &metainternalscheme.Codecs,
+		aggregatorscheme.Scheme:       &aggregatorscheme.Codecs,
+		apiextensionsapiserver.Scheme: &apiextensionsapiserver.Codecs,
+	}
+
+	for scheme, factory := range codecs {
+		original := *factory // shallow copy of original value
+		tb.Cleanup(func() { *codecs[scheme] = original })
+		*codecs[scheme] = serializer.NewCodecFactory(scheme, serializer.WithSerializer(newCBORSerializerInfo))
+	}
+}
+
+// AssertRequestResponseAsCBOR returns a transport.WrapperFunc that will report a test error if a
+// non-empty request or response body contains data that does not appear to be CBOR-encoded.
+func AssertRequestResponseAsCBOR(t testing.TB) transport.WrapperFunc {
+	recognizer := cbor.NewSerializer(runtime.NewScheme(), runtime.NewScheme())
+
+	unsupportedPatchContentTypes := sets.New(
+		"application/json-patch+json",
+		"application/merge-patch+json",
+		"application/strategic-merge-patch+json",
+	)
+
+	return func(rt http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(request *http.Request) (*http.Response, error) {
+			if request.Body != nil && !unsupportedPatchContentTypes.Has(request.Header.Get("Content-Type")) {
+				requestbody, err := io.ReadAll(request.Body)
+				if err != nil {
+					t.Error(err)
+				}
+				recognized, _, err := recognizer.RecognizesData(requestbody)
+				if err != nil {
+					t.Error(err)
+				}
+				if len(requestbody) > 0 && !recognized {
+					t.Errorf("non-cbor request: 0x%x", requestbody)
+				}
+				request.Body = io.NopCloser(bytes.NewReader(requestbody))
```

Comment:
please open a PR based on this one that `t.Error` here and in the cleanup below so it's obvious that these paths are being hit.

Reply:
Opened https://github.com/kubernetes/kubernetes/pull/128295 and kicked off integration tests. I intentional fail to enable CBOR properly in that PR. Waiting for the presubmit output, but locally I see a lot of this:

```
cbor.go:110: non-cbor request: 0x7b226b696e64223a2244656c6574654f7074696f6e73222c2261706956657273696f6e223a227631222c226772616365506572696f645365636f6e6473223a302c2270726f7061676174696f6e506f6c696379223a224261636b67726f756e64227d0a
cbor.go:136: non-cbor response: 0x7b226b696e64223a22537461747573222c2261706956657273696f6e223a227631222c226d65746164617461223a7b7d2c22737461747573223a224661696c757265222c226d657373616765223a22766f6c756d6561747472696275746573636c61737365732e73746f726167652e6b38732e696f205c22766163325c22206e6f7420666f756e64222c22726561736f6e223a224e6f74466f756e64222c2264657461696c73223a7b226e616d65223a2276616332222c2267726f7570223a2273746f726167652e6b38732e696f222c226b696e64223a22766f6c756d6561747472696275746573636c6173736573227d2c22636f6465223a3430347d0a
```

They're both JSON, the first is:
```json
{"kind":"DeleteOptions","apiVersion":"v1","gracePeriodSeconds":0,"propagationPolicy":"Background"}
```
and the second is:
```json
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"volumeattributesclasses.storage.k8s.io \"vac2\" not found","reason":"NotFound","details":{"name":"vac2","group":"storage.k8s.io","kind":"volumeattributesclasses"},"code":404}
```

Reply:
great.